include/ignition/gui/plugins/KeyPublisher.hh

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef IGNITION_GUI_KEYPUBLISHER_HH_
+#define IGNITION_GUI_KEYPUBLISHER_HH_
+
+#include <ignition/gui/qt.h>
+#include <ignition/gui/Plugin.hh>
+#include <ignition/transport/Node.hh>
+
+
+namespace ignition
+{
+namespace gui
+{
+  class KeyPublisher : public ignition::gui::Plugin  //inherited from Qobject
+  {
+    Q_OBJECT
+
+    public: KeyPublisher();    //constructor
+    public: virtual ~KeyPublisher();   //destructor
+    
+    public: virtual void LoadConfig(const tinyxml2::XMLElement *_pluginElem) override;
+    protected: bool eventFilter(QObject *_obj, QEvent *_event) override;
+
+    //Publish message
+    public: ignition::transport::Node node;
+    public: std::string topic = "/gazebo/keyboard/keypress";
+    public: ignition::transport::Node::Publisher pub ;
```

Comment:
We've been using the PIMPL pattern for all plugins. It's helpful in case the headers are installed, which is not currently the case, but if we just follow the pattern it will be easier if we ever install it.

You can see examples on the other plugins. They have all private members inside a private class, i.e.

https://github.com/ignitionrobotics/ign-gui/blob/9da3dec08504774e7ebe9d446b0ab6acf38d299b/include/ignition/gui/plugins/Publisher.hh#L146-L148

Let me know if you have questions!